src/lib/coda_lib/coda_lib.ml

```diff
@@ -479,28 +479,39 @@ module Make (Inputs : Inputs_intf) = struct
 
   let propose_keypair t = t.propose_keypair
 
+  let best_tip t = Transition_frontier.best_tip t.transition_frontier
+
   let best_ledger_builder t =
-    failwith
-      "TODO: Use transition frontier to get best lb out; you'll need to \
-       update the signature"
+    Transition_frontier.Breadcrumb.staged_ledger (best_tip t)
+    |> Transition_frontier.hack_temporary_ledger_builder_of_staged_ledger
 
   let best_protocol_state t =
-    failwith
-      "TODO: Use transition frontier to get best lb out; you'll need to \
-       update the signature"
-
-  let best_tip t =
-    failwith
-      "TODO: Use transition frontier to get best lb out; you'll need to \
-       update the signature"
-
-  let get_ledger t lh =
-    failwith
-      "TODO: Use transition frontier to find an arbitrary ledger based on a \
-       hash"
+    Transition_frontier.Breadcrumb.transition_with_hash (best_tip t)
+    |> With_hash.data |> External_transition.protocol_state
 
   let best_ledger t = Ledger_builder.ledger (best_ledger_builder t)
 
+  let get_ledger t ledger_builder_hash =
+    match
+      List.find_map (Transition_frontier.all_breadcrumbs t.transition_frontier)
```

Comment:
inefficient but whatever for now

Reply:
I'm fine with this only since `get_ledger` is a hack anyway. Long term, we need to remove it.